app/controllers/impersonate_controller.rb

```diff
@@ -0,0 +1,52 @@
+class ImpersonateController < ApplicationController
+  after_action :verify_authorized
+
+  # Session parameter name for impersonation id
+  IMPERSONATE_USER_PARAM = 'impersonate_user_id'.freeze
+
+  # GET /impersonate
+  def index
+    authorize :impersonate
+    @users = User.all
```

Comment:
I don't think you need this line? Querying the db for all users but then reassigning